cms/djangoapps/contentstore/management/commands/delete_course.py

```diff
@@ -50,7 +53,9 @@ def add_arguments(self, parser):
 
     def handle(self, *args, **options):
         try:
-            course_key = CourseKey.from_string(options['course_key'])
+            # a course key may have unicode chars in it
+            course_key = unicode(options['course_key'], 'utf8')
+            course_key = CourseKey.from_string(course_key)
```

Comment:
@noraiz-anwar , I'm guessing `from_string` should work with unicode course keys. Does it not?

Reply:
`from_string` throws `UnicodeDecodeError` for  unicode course keys. Here is the complete traceback.
``` python
  Traceback (most recent call last):
  File "manage.py", line 116, in <module>
    execute_from_command_line([sys.argv[0]] + django_args)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/management/commands/delete_course.py", line 53, in handle
    course_key = CourseKey.from_string(options['course_key'])
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/opaque_keys/__init__.py", line 193, in from_string
    namespace, rest = cls._separate_namespace(serialized)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/opaque_keys/__init__.py", line 213, in _separate_namespace
    namespace, __, rest = serialized.partition(cls.NAMESPACE_SEPARATOR)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 13: ordinal not in range(128) ```

Reply:
Oh, yikes. Good thing we're getting ahead of this bug then!

Reply:
It seems to work when I run it directly in a shell

```python
In [1]: from opaque_keys.edx.keys import CourseKey

In [2]: CourseKey.from_string(u"course-v1:ørg+course+run")
Out[2]: CourseLocator(u'\xf8rg', u'course', u'run', None, None)
```

Reply:
Actually we need to pass unicode course key in `from_string`. That is what I am doing in this fix. But as we pass course_key as command line argument, course key is taken as str despite the fact that it contains unicode chars. We need to explicitly declare it unicode so `from_string` can parse it as expected.

Reply:
ah, I see :)